apps/ensnode/src/handlers/Registrar.ts

```diff
@@ -76,8 +77,15 @@ export const makeRegistrarHandlers = (ownedName: OwnedName) => {
 
       const node = makeSubnodeNamehash(ownedNameNode, labelhash);
 
-      // TODO: materialze labelName via rainbow tables ala Registry.ts
-      const labelName = undefined;
+      // materialze labelName via rainbow tables ala Registry.ts
+      // https://github.com/ensdomains/ens-subgraph/blob/c8447914e8743671fb4b20cffe5a0a97020b3cee/src/ethRegistrar.ts#L56-L61
+      const label = await labelByHash(labelhash);
+      const canIndexLabel = isLabelIndexable(label);
+      // NOTE: undefined type is helpful for passing to drizzle, which
+      // will ignore the input rather that set the value in the db to `NULL`
+      // if we were to return `null`.
+      const labelName = canIndexLabel ? label : undefined;
+      const name = canIndexLabel ? `${label}.${ownedName}` : undefined;
```

Comment:
@shrugs please share your thoughts 🚀

Reply:
Replicated the same behaviour in a slightly different syntax. The bottom line is that we don't want to update `labelName` & `name` on an indexed `domain` entity (or `labelName` on an indexed `registration` entity) if the healed label is not indexable.

The original ENS Subgraph implementation:
https://github.com/ensdomains/ens-subgraph/blob/c8447914e8743671fb4b20cffe5a0a97020b3cee/src/ethRegistrar.ts#L56-L61
```ts
let labelName = ens.nameByHash(label.toHexString());
if (checkValidLabel(labelName)) {
  domain.labelName = labelName;
  domain.name = labelName! + ".eth";
  registration.labelName = labelName;
}
```